accounts/abi/reflect.go

```diff
@@ -24,6 +24,64 @@ import (
 	"strings"
 )
 
+// ToStruct converts a interface of a runtime type into a interface of the
+// given type
+// e.g. turn
+// var fields []reflect.StructField
+// fields = append(fields, reflect.StructField{
+// 		Name: "X",
+//		Type: reflect.TypeOf(new(big.Int)),
+//		Tag:  reflect.StructTag("json:\"" + "x" + "\""),
+// }
+// into
+// type TupleT struct { X *big.Int }
+func ToStruct(in interface{}, proto interface{}) interface{} {
+	inType, protoType := reflect.TypeOf(in), reflect.TypeOf(proto)
+	switch {
+	case inType.ConvertibleTo(protoType):
+		return reflect.ValueOf(in).Convert(protoType).Interface()
+	case inType.Kind() == reflect.Struct:
+		if err := copyStruct(proto, in); err != nil {
+			return nil
+		}
+		return proto
+	case inType.Kind() == reflect.Array:
```

Comment:
`ToStruct` can take an array as an input and returns an array, i.e. not a struct?

Reply:
Yep, will change naming here